core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorMessageDto.java

```diff
@@ -2,26 +2,53 @@
 
 import java.util.Collection;
 import java.util.Map;
+import java.util.Set;
 
 import lombok.Builder;
 import lombok.Value;
 
 import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Multimaps;
 import com.google.common.collect.SetMultimap;
+import com.google.common.collect.Sets;
 
 @Value
 @Builder(toBuilder = true)
 public class ErrorMessageDto
 {
-    private final Map<String, Collection<ErrorDto>> errors;
+    Map<String, Collection<ErrorDto>> errors;
+    Map<String, Object> globalParameters;
 
     public ErrorMessageDto combineWith(ErrorMessageDto other)
     {
-        SetMultimap<String, ErrorDto> combined = HashMultimap.create();
+        return toBuilder().errors(combineErrors(other).asMap())
+            .globalParameters(combineGlobalParameters(other))
+            .build();
+    }
 
+    private SetMultimap<String, ErrorDto> combineErrors(ErrorMessageDto other)
+    {
+        SetMultimap<String, ErrorDto> combined = HashMultimap.create();
         errors.forEach(combined::putAll);
-        other.getErrors().forEach(combined::putAll);
+        other.getErrors()
+            .forEach(combined::putAll);
+        return Multimaps.unmodifiableSetMultimap(combined);
+    }
 
-        return toBuilder().errors(combined.asMap()).build();
+    private Map<String, Object> combineGlobalParameters(ErrorMessageDto other)
+    {
+        Map<String, Object> otherGlobalParameters = other.getGlobalParameters();
+
+        Set<String> combinedKeys = Sets.union(globalParameters.keySet(), otherGlobalParameters.keySet());
```

Comment:
I was actually thinking along the lines of streaming both key sets, then using `Stream.concat`. But this is acceptable as well.